pkg/kubelet/network/cni/cni.go

```diff
@@ -53,14 +59,18 @@ type cniNetwork struct {
 
 func probeNetworkPluginsWithVendorCNIDirPrefix(pluginDir, vendorCNIDirPrefix string) []network.NetworkPlugin {
 	configList := make([]network.NetworkPlugin, 0)
-	network, err := getDefaultCNINetwork(pluginDir, vendorCNIDirPrefix)
-	if err != nil {
-		return configList
-	}
-	return append(configList, &cniNetworkPlugin{
-		defaultNetwork: network,
+	plugin := &cniNetworkPlugin{
+		defaultNetwork: nil,
 		execer:         utilexec.New(),
-	})
+	}
+
+	plugin.syncNetworkConfig(pluginDir, vendorCNIDirPrefix)
+	// sync network config from pluginDir periodically to detect network config updates
+	go wait.Forever(func() {
+		plugin.syncNetworkConfig(pluginDir, vendorCNIDirPrefix)
+	}, 10*time.Second)
```

Comment:
nit: Do we have some common place to store loop intervals and timeouts?
Then this could be moved there... but that's not something that have to be done in this PR.